CI: Execute checkstyle CI step on each run

cpeeyush opened the pull request:
## What this PR changes/adds

checkstyle check was not getting executed in downstram forks as it was configured to run on PR against `main` branch only. This PR removes this restriction so that checkstyle step gets execute on every CI run and also marks checkstyle step as failure in case of any checkstyle errors.



## Why it does that

Detect checkstyle related errors earlier e.g. downstream forks.

## Linked Issue(s)

Relates #314 

## Checklist

- [ ] added appropriate tests?
- [x] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

cpeeyush commented:
checkstyle step executed successfully https://github.com/agera-edc/DataSpaceConnector/runs/7147503069?check_suite_focus=true

algattik reviewed (changes requested):
How does that work with the `reporter: 'github-pr-check'`?

It seems to more more that the triggers are inconsistent
```

  push:
  pull_request:
    branches: [ main ]
```

why run on push to any branch, but on PR to main only? Should we rather than the workflow on PR to any branch (and perhaps not on push)?

cpeeyush commented:
> How does that work with the `reporter: 'github-pr-check'`?
> 
> It seems to more more that the triggers are inconsistent
> 
> ```
> 
>   push:
>   pull_request:
>     branches: [ main ]
> ```
> 
> why run on push to any branch, but on PR to main only? Should we rather than the workflow on PR to any branch (and perhaps not on push)?

Good catch updated action reporter to github-check.

https://github.com/agera-edc/DataSpaceConnector/runs/7147703763

Regarding the other change you suggested to remove push trigger I am not sure as I am not aware about the original purpose of it. So maybe we can discuss it with upstream devs first.

algattik commented:
> Good catch updated action reporter to github-check.

won't that lead to reports not being published as PR comments anymore? I'd suggest keeping config as-is

cpeeyush commented:
> > Good catch updated action reporter to github-check.
> 
> won't that lead to reports not being published as PR comments anymore? I'd suggest keeping config as-is

@algattik Report is still getting published as PR Comment. Here you can see I have added an error on purpose to verify this one I can see a PR comment from checkstyle action https://github.com/agera-edc/DataSpaceConnector/pull/316/commits/a18058da3187ce365eba1bb6cfcc757c78ceb82c

As per its doc it says
_It's basically same as -reporter=github-pr-check except it works not only for Pull Request but also for commit._

algattik commented:
What about the comment `          # Filtering does not work on github-check, needs github-pr-check`?

cpeeyush commented:
> What about the comment ` # Filtering does not work on github-check, needs github-pr-check`?

@algattik Updated. Removed main branch PR trigger only. So now checkstyle check will run on every PR and not just PRs against main branch.
Kept the config as-is before.

algattik commented:
> But now entire codebase follows checkstyle rules and no pre-existing issues are there so this filtering is not needed.

Note that the default filter is `added`. If you want to disable filtering you will have to set another value.

cpeeyush commented:
> > But now entire codebase follows checkstyle rules and no pre-existing issues are there so this filtering is not needed.
> 
> Note that the default filter is `added`. If you want to disable filtering you will have to set another value.

Reverted filter changes. Just updated reporter to run it on commits as well and not only on PRs. Added fail-on-error to fail checkstyle step in case of failue.
